### DISCUSSION TITLE
Phase field (crystal) question

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#17794] [Q&A Modules: Phase field] [answered]
I have a request from some colleagues to model something, but I am unsure where to look for the tools:

The system starts initially amorphous and has some small perhaps randomly placed nucleation centers that form a specific (cubic) crystalline phase. In the interface region, there may be some pyrochlore crystallites that comprise an "intermediate" phase.
The nucleation centers essentially grow until they fill the computational domain and we get a number of different grains/grain sizes based on the temperature profile the evolution sees. 

Is such a model implemented in MOOSE, the modules, or any external apps? Is this "phase field crystal"?

**Comment by aeslaughter** [comment]
@SudiptaBiswas Can you please help?

**Comment by laagesen** [comment]
I wouldn't call this phase-field crystal. PFC models represent the local atomic structure with a continuous order parameter that is analogous to local density. The length scale is much smaller than typical phase-field models because PFC attempts to resolve the local atomic structure, whereas in phase-field modeling we coarse-grain the atomic structure.

What you are describing sounds more like a phase transformation problem. It seems similar to a solidification problem, except that rather than transforming from the liquid phase to multiple grains of a solid phase, you are transforming from an amorphous phase to multiple grains of a crystalline (solid) phase. In both situations, there is a driving force in the free energy that is causing the phase transformation. I'm not sure offhand how the intermediate phase in the interface would affect things.

So setting aside the intermediate phase for the moment, to model this you would want to be able to represent the amorphous phase (a single order parameter for this should suffice) and multiple grains of the crystalline phase. One question I have is whether there is any difference in chemical composition between the amorphous and crystalline phases? (I am guessing not but just thought I would double check.)

**Reply by mangerij** [reply]
Yeah, there is no difference in chemical composition, the material just has three different phases that evolve from amorphous to a full grain structure via small nucleation centers. The tertiary phase can be ignored for now. 

So where are these "phase transformation" models in the modules? Is there an input file I can work off of? 

thanks!

**Comment by SudiptaBiswas** [comment]
I don't think we have the capability to set up the interface as a third material. I agree with Larry, the two-phase problem can be handled with existing phase transformation models.

**Comment by laagesen** [comment, marked as answer]
So we don't have any examples that do exactly this situation in the repo now. It is probably closest to the solidification examples here:

https://github.com/idaholab/moose/tree/next/modules/phase_field/examples/anisotropic_interfaces

I would probably start with the grand potential formulation for a binary alloy here:

https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/anisotropic_interfaces/GrandPotentialSolidification.i

So for this problem, you won't need the evolution equations for chemical potential, since you aren't tracking a solute species. You also likely don't need interfacial energy anisotropy. What you are wanting to do is actually pretty similar to a grain growth model, except that there is an energetic driving force that makes the crystalline phase more energetically favorable than the amorphous phase. So from this example, you would need to add multiple order parameters for the crystalline phase rather than the single phase as in this example. 

For more details on the formulation of the models described above please see:
https://journals.aps.org/pre/abstract/10.1103/PhysRevE.98.023309
Again you won't need solute species evolution so there is a lot of stuff you won't need, but I think this is the most straightforward way to go from what is currently in the MOOSE repo. You may be better off writing your own evolutions equations by simplifying what is in the paper, then using existing kernels and materials to implement it. That should be doable with no new code.
